zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -830,6 +830,29 @@ function ParagraphCtrl ($scope, $rootScope, $route, $window, $routeParams, $loca
       // autocomplete on 'ctrl+.'
       $scope.editor.commands.bindKey('ctrl-.', 'startAutocomplete')
 
+      // Show autocomplete on tab
+      $scope.editor.commands.addCommand({
+        name: 'tabAutocomplete',
+        bindKey: {
+          win: 'tab',
+          mac: 'tab',
+          sender: 'editor|cli'
+        },
+        exec: function(env, args, request) {
+          let iColumnPosition = $scope.editor.selection.getCursor().column
+
+          // If user has pressed tab on first line char or if editor mode is %md, keep existing behavior
+          // If user has pressed tab anywhere in between and editor mode is not %md, show autocomplete
+          if (iColumnPosition && $scope.paragraph.config.editorMode !== 'ace/mode/markdown') {
```

Comment:
"Interpreter setting" in zeppelin is a bit confusing because we use this words in two different places. One place is in 'Interpreter' menu, we can create interpreter settings and configure them. Another place is "Interpreter-setting.json" in each interpreter's source tree, which has default values for property and some immutable configurations.

What I meant was the immutable configuration in the second one.
For example markdown interpreter has [immutable 'editor' configuration](https://github.com/apache/zeppelin/blob/master/markdown/src/main/resources/interpreter-setting.json#L15-L17), like

```
    "editor": {
      "language": "markdown",
      "editOnDblClick": true
```


And editor property can be accessed from the front end, like [here](https://github.com/apache/zeppelin/blob/340b326d47f3e07b7d98c2760b41c10424bfbb30/zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js#L447) or [here](https://github.com/apache/zeppelin/blob/340b326d47f3e07b7d98c2760b41c10424bfbb30/zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js#L961).

So, we can make Tab or any other key as a default key for completion, but developers of each interpreter should able to override best completion key for each interpreter in the source code of interpreter.

I haven't thought deeply about each user override shortcut of completion key. But once Zeppelin allow user customizing shortcut, I think it will be not only completion key, but all other shortcut keys.